pandas/core/arrays/categorical.py

```diff
@@ -370,6 +373,224 @@ def __init__(
+        A column whose header is NA will be dropped:
+        any row containing a NA value will be uncategorised.
+
+        Parameters
+        ----------
+        dummies : DataFrame
+            dtypes of columns with non-NA headers must be coercible to bool.
+            Sparse dataframes are not supported.
+        ordered : bool
+            Whether or not this Categorical is ordered.
+        prefix : optional str
+            Only take columns whose names are strings starting
+            with this prefix and ``prefix_sep``,
+            stripping those elements from the resulting category names.
+        prefix_sep : str, default "_"
+            If ``prefix`` is not ``None``, use as the separator
+            between the prefix and the final name of the category.
+        fillna : optional bool, default None
+            How to handle NA values. If ``True`` or ``False``, NA is filled with that value.
+            If ``None``, raise a ValueError if there are any NA values.
+
+        Raises
+        ------
+        ValueError
+            If a sample belongs to >1 category
+
+        Returns
+        -------
+        Categorical
+
+        Examples
+        --------
+        >>> simple = pd.DataFrame(np.eye(3), columns=["a", "b", "c"])
+        >>> Categorical.from_dummies(simple)
+        [a, b, c]
+        Categories (3, object): [a, b, c]
+
+        >>> nan_col = pd.DataFrame(np.eye(4), columns=["a", "b", np.nan, None])
+        >>> Categorical.from_dummies(nan_col)
+        [a, b, NaN, NaN]
+        Categories (2, object): [a, b]
+
+        >>> nan_cell = pd.DataFrame(
+        ...     [[1, 0, np.nan], [0, 1, 0], [0, 0, 1]],
+        ...     columns=["a", "b", "c"],
+        ... )
+        >>> Categorical.from_dummies(nan_cell)
+        [NaN, b, c]
+        Categories (3, object): [a, b, c]
+
+        >>> multi = pd.DataFrame(
+        ...     [[1, 0, 1], [0, 1, 0], [0, 0, 1]],
+        ...     columns=["a", "b", "c"],
+        ... )
+        >>> Categorical.from_dummies(multi)
+        Traceback (most recent call last):
+            ...
+        ValueError: 1 record(s) belongs to multiple categories: [0]
+        """
+        from pandas import Series
+
+        copied = False
+        to_drop = dummies.columns[isna(dummies.columns.values)]
+        if len(to_drop):
+            dummies = dummies.drop(columns=to_drop)
+            copied = True
+
+        if prefix is None:
+            cats = dummies.columns
+        else:
+            pref = prefix + (prefix_sep or "")
+            cats = []
+            to_keep = []
+            for c in dummies.columns:
+                if isinstance(c, str) and c.startswith(pref):
+                    to_keep.append(c)
+                    cats.append(c[len(pref) :])
+            dummies = dummies[to_keep]
+
+        df = dummies.astype("boolean")
+        if fillna is not None:
+            df = df.fillna(fillna, inplace=copied)
+
+        row_totals = df.sum(axis=1, skipna=False)
```

Comment:
why skipna?

Reply:
If there is no explicit `fillna` policy given, and there are still NA values in the data, I'd prefer to raise an error rather than silently pass bunk data through. Therefore nans should not be skipped in this step so that they can be checked for in the next line.